packages/lit-ssr/src/test/integration/client/basic_test.ts

```diff
@@ -0,0 +1,331 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import '@open-wc/testing';
+
+import {tests} from '../tests/basic.js';
+import {render} from 'lit-html';
+import {hydrate} from 'lit-html/hydrate.js';
+import {hydrateShadowRoots} from 'template-shadowroot/template-shadowroot.js';
+import {SSRExpectedHTML} from '../tests/ssr-test.js';
+
+const assert = chai.assert;
+
+const assertTemplate = document.createElement('template');
+
+/**
+ * Removes comments, normalizes text content, and normalizes style attribute
+ * serialization.
+ *
+ * Although semantic-dom-diff removes comments, it does not normalize
+ * textContent after removing them, so differences in comments remain
+ * semantically meaningful; we currently use different comment marker strategies
+ * between client and server, so this inures us to those differences.
+ */
+const normalize = (node: Node | null) => {
+  if (node === null) {
+    return;
+  }
+  let nextSibling = node.nextSibling;
+  // Remove comments
+  if (node.nodeType === node.COMMENT_NODE) {
+    node.parentNode!.removeChild(node);
+  }
+  // Normalize text content (we don't use node.normalize, because
+  // it doesn't work on IE)
```

Comment:
Can we just polyfill this? Maybe in the webcomponents platform polyfills?

Reply:
We need to remove all comment nodes first before normalizing the text content, and since you can't query for comments to remove them, you need to walk anyway, and at that point it's just not that big of deal to do the text node collapsing.

Reply:
ok